daml-script/converter/src/main/scala/Converter.scala

```diff
@@ -32,7 +32,7 @@ private[daml] object Converter {
     val fieldNames = fields.view.map { case (n, _) => Name.assertFromString(n) }.to(ImmArray)
     val args =
       new util.ArrayList[SValue](fields.map({ case (_, v) => v }).asJava)
-    SRecord(ty, fieldNames, args)
+    SRecord(ty, fieldNames, args) // better to construct SRecord directly from Map
```

Comment:
If you're not doing this now, make this a TODO

Reply:
okay